test/router.test.ts

```diff
@@ -0,0 +1,199 @@
+import poolData from "./test-pools.json";
+import { ThalaswapRouter } from "../src";
+import { expect, test, jest } from "bun:test";
+
+const coins = poolData.coins;
+const pools = poolData.pools.map((pool: any) => {
+  return {
+    ...pool,
+    asset0: coins[pool.asset0],
+    asset1: coins[pool.asset1],
+    asset2: pool.asset2 ? coins[pool.asset2] : undefined,
+    asset3: pool.asset3 ? coins[pool.asset3] : undefined,
+  };
+});
+
+const mockPoolDataClient = {
+  getPoolData: jest.fn(() => {
+    return new Promise((resolve) =>
+      resolve({
+        pools: pools,
+        coins: coins,
+      }),
+    );
+  }),
+};
+
+const router = new ThalaswapRouter("example-url");
+router.setPoolDataClient(mockPoolDataClient as any);
+
+test("Exact input 1 hop", async () => {
+  const startToken =
+    "0xf22bede237a07e121b56d91a491eb7bcdfd1f5907926a9e58338f964a01b17fa::asset::USDC";
+  const endToken = "0x1::aptos_coin::AptosCoin";
+  const amountIn = 1000;
+
+  const route = await router.getRouteGivenExactInput(
+    startToken,
+    endToken,
+    amountIn,
+    1,
+  );
+  console.log(
+    `From: ${startToken}, To: ${endToken}, AmountIn: ${amountIn}, Optimal Route:`,
+    route,
+  );
+  expect(route!.path.length).toBe(1);
+  expect(route!.path[0].from === startToken).toBe(true);
+  expect(route!.path[0].to === endToken).toBe(true);
+});
+
+test("Exact input 2 hop", async () => {
+  const startToken =
+    "0xf22bede237a07e121b56d91a491eb7bcdfd1f5907926a9e58338f964a01b17fa::asset::WETH";
+  const endToken = "0x1::aptos_coin::AptosCoin";
+  const amountIn = 1;
+
+  const route = await router.getRouteGivenExactInput(
+    startToken,
+    endToken,
+    amountIn,
+    2,
+  );
+  console.log(
+    `From: ${startToken}, To: ${endToken}, AmountIn: ${amountIn}, Optimal Route:`,
+    route,
+  );
+  expect(route!.path.length).toBe(2);
```

Comment:
let's also assert amountOut and priceImpactPercentage, so that if we refactor code we know if it actually changes the behavior. this applies to all test cases